fix #701: Redirect user to the console when he tries to edit phone number on profile page

jumarko opened the pull request:
Since phone change requires sms verification, we don't want user let to edit it directly on profile page.
Instead, we redirect him to the Console automatically issuing :phone command.

jumarko commented on src/status_im/profile/screen.cljs:
```diff
@@ -155,10 +163,13 @@
-                                     (label :t/not-specified))}]
+                                     (label :t/not-specified))
+                        ;; TODO: should this be changed?
+                        :on-press  navigate-to-phone-change}]
```
Not sure if this should be changed - where is the profile view used?

alwx replied:
The profile view you are talking about is another user's profile. You can access it by opening a chat with some user and pressing "Profile" in popup menu after that.
So the answer is no, we don't need to navigate to console chat in this case.

jumarko commented on src/status_im/profile/screen.cljs:
```diff
@@ -113,6 +114,13 @@
+  []
+  (dispatch [:navigate-to :chat console-chat-id])
+  (dispatch [:set-chat-command :phone])
+  )
```
Is this acceptable or should I create new event like `:phone-number-change-requested` and dispatch these two events in the separate handler?

alwx replied:
IMO this is acceptable, but it is better to move it to another handler just for clarity and readability

alwx commented:
Thank you for your contribution!
We don't currently have unit tests, and I don't see any need for documentation changes.

jumarko commented:
I removed the change of "other user profile" in fixup commit.

rasom commented on src/status_im/profile/screen.cljs:
```diff
@@ -155,10 +163,11 @@
                         :value     (if (and phone (not (str/blank? phone)))
                                      (format-phone-number phone)
-                                     (label :t/not-specified))}]
+                                     (label :t/not-specified))
```
```
----  Could not Read  src/status_im/profile/screen.cljs   line:171  column:36  ----

  Unmatched delimiter ]

  169                                       (format-phone-number phone)
  170                                       (label :t/not-specified))
  171       [view st/underline-container]]
                                          ^--- Unmatched delimiter ]
  172
  173      (when address
  174        [view st/profile-property

----  Reader Error : Please see src/status_im/profile/screen.cljs  ----
```

jumarko commented:
@rasom @alwx I've updated the pull request - it should compile now.
I also changed the approach a little bit by introducing new event :phone-number-change-requested which is dispatched when user clicks on phone field on profile screen.

rasom commented:
Thanks for updates, @jumarko !
